### ISSUE TITLE
Deepin 15.6/15.7 freezes after rearranging icons on tray (X error BadWindow)

### ISSUE THREAD

**Issue by igormcoelho** [linuxdeepin/dde-dock#114]
My desktop is constantly freezing when I try to rearrange icons (on efficient mode). In fact, I took some time to realize this was the reason behind the constant freezing (that seemed random to me). These are journalctl messages:
```
ago 26 15:47:07 imcoelho-PC dbus-daemon[2877]: [session uid=1000 pid=2877] Activating service name='com.deepin.dde.Dock' requested by ':1.6' (uid=1000 pid=2858 comm="/usr/bin/startdde " label="unconfined")
ago 26 15:47:07 imcoelho-PC com.deepin.dde.Dock[2877]: load translate "/usr/share/dtkwidget/translations/dtkwidget2_pt_BR"
ago 26 15:47:07 imcoelho-PC com.deepin.dde.Dock[2877]: load translate "/usr/share/dde-dock/translations/dde-dock_pt_BR"
ago 26 15:47:07 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:07.701 [Debug  ] [                                                         0]
ago 26 15:47:07 imcoelho-PC com.deepin.dde.Dock[2877]: dde-dock startup
ago 26 15:47:07 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:07.753 [Debug  ] [                                                         0] void DockSettings::updateForbidPostions()
ago 26 15:47:07 imcoelho-PC dbus-daemon[2877]: [session uid=1000 pid=2877] Successfully activated service 'com.deepin.dde.Dock'
ago 26 15:47:07 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:07.979 [Debug  ] [                                                         0] init plugin:  "datetime"
ago 26 15:47:07 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:07.979 [Debug  ] [                                                         0] load plugins item:  "datetime" "" DatetimeWidget(0x559d5f39c210)
ago 26 15:47:07 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:07.980 [Debug  ] [                                                         0] init plugin finished:  "datetime"
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.115 [Debug  ] [                                                         0] screen info:  QPoint(0,725) QRect(0,725 1366x43)
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.386 [Debug  ] [                                                         0] screen info:  QPoint(0,725) QRect(0,725 1366x43)
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.480 [Debug  ] [diskmountplugin.cpp  DiskMountPlugin::DiskMountPlugin    44] ===============init=============
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.480 [Debug  ] [                                                         0] init plugin:  "disk-mount"
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.480 [Debug  ] [dapplication.cpp     Dtk::Widget::DApplicationPrivate::loadTranslator 255] load translate "/usr/share/dtkwidget/translations/dtkwidget2_pt_BR"
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.481 [Debug  ] [dapplication.cpp     Dtk::Widget::DApplicationPrivate::loadTranslator 255] load translate "/usr/share/dde-file-manager/translations/dde-file-manager_pt_BR"
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.481 [Debug  ] [diskmountplugin.cpp  DiskMountPlugin::init               66] true
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.481 [Debug  ] [diskmountplugin.cpp  DiskMountPlugin::init               68] ===============init==proxyInter===========
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.575 [Debug  ] [diskmountplugin.cpp  DiskMountPlugin::diskCountChanged   164] 1 false
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.575 [Debug  ] [                                                         0] load plugins item:  "disk-mount" "" DiskPluginItem(0x559d5f3a54c0)
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.578 [Debug  ] [                                                         0] init plugin finished:  "disk-mount"
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.982 [Debug  ] [                                                         0] init plugin:  "trash"
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.982 [Debug  ] [dapplication.cpp     Dtk::Widget::DApplicationPrivate::loadTranslator 255] load translate "/usr/share/dtkwidget/translations/dtkwidget2_pt_BR"
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.982 [Debug  ] [dapplication.cpp     Dtk::Widget::DApplicationPrivate::loadTranslator 255] load translate "/usr/share/dde-file-manager/translations/dde-file-manager_pt_BR"
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.982 [Debug  ] [trashplugin.cpp      TrashPlugin::init                   54] true
ago 26 15:47:08 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:08.982 [Debug  ] [                                                         0] init plugin finished:  "trash"
ago 26 15:47:09 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:09.479 [Debug  ] [                                                         0] init plugin:  "keyboard"
ago 26 15:47:09 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:09.481 [Debug  ] [                                                         0] init plugin finished:  "keyboard"
ago 26 15:47:09 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:09.981 [Debug  ] [                                                         0] init plugin:  "network"
ago 26 15:47:09 imcoelho-PC com.deepin.dde.Dock[2877]: /usr/bin/proxychains4
ago 26 15:47:09 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:09.988 [Debug  ] [                                                         0] init plugin finished:  "network"
ago 26 15:47:09 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:09.992 [Warning] [dthememanager.cpp    Dtk::Widget::DThemeManagerPrivate::registerWidget 234] can not load qss with out filename Dtk::Widget::DDoubleSpinBox(0x559d5f4c9bd0)
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.030 [Warning] [                                                         0] QMetaObject::invokeMethod: No such method WirelessList::loadAPList()
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.032 [Debug  ] [                                                         0] load plugins item:  "network" "/org/freedesktop/NetworkManager/Devices/2" WiredItem(0x559d5f4037f0)
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.032 [Debug  ] [                                                         0] load plugins item:  "network" "/org/freedesktop/NetworkManager/Devices/3" WirelessItem(0x559d5f46a2e0)
ago 26 15:47:10 imcoelho-PC daemon/dock[3132]: dock_manager_xevent.go:154: WindowError{code=3, seq=779, major=2, minor=0, bad=27263164}
ago 26 15:47:10 imcoelho-PC bamfdaemon[3267]: Got an X error: BadWindow (invalid Window parameter)
ago 26 15:47:10 imcoelho-PC bamfdaemon[3267]: Got an X error: BadWindow (invalid Window parameter)
ago 26 15:47:10 imcoelho-PC bamfdaemon[3267]: Got an X error: BadWindow (invalid Window parameter)
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.480 [Debug  ] [                                                         0] init plugin:  "shutdown"
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.481 [Debug  ] [                                                         0] load plugins item:  "shutdown" "shutdown" PluginWidget(0x559d5f636280)
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.483 [Debug  ] [                                                         0] load plugins item:  "shutdown" "power" PowerStatusWidget(0x559d5f5cedf0)
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.483 [Debug  ] [                                                         0] init plugin finished:  "shutdown"
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.980 [Debug  ] [                                                         0] init plugin:  "sound"
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.983 [Debug  ] [                                                         0] load plugins item:  "sound" "" SoundItem(0x559d5f607c80)
ago 26 15:47:10 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:10.983 [Debug  ] [                                                         0] init plugin finished:  "sound"
ago 26 15:47:11 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:11.481 [Debug  ] [                                                         0] init plugin:  "system-tray"
ago 26 15:47:11 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:11.481 [Debug  ] [                                                         0] init plugin finished:  "system-tray"
ago 26 15:47:11 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:11.484 [Debug  ] [                                                         0] load plugins item:  "system-tray" "window:56623108" XWindowTrayWidget(0x7f8c7401e0d0)
ago 26 15:47:11 imcoelho-PC com.deepin.dde.Dock[2877]: 2018-08-26, 15:47:11.486 [Debug  ] [                                                         0] delay load 0 "keybord_layout" IndicatorTray(0x559d5f556890)
```

I think the bug reason must be this `Got an X error: BadWindow (invalid Window parameter)`

The easiest way to reproduce the bug is:
- Open a terminal (Ctrl+Alt+T)
- Maximize the terminal
- Try to move the try icon to inside the terminal window
- repeate this many times, and in a few tries the system will freeze

It happened in many other situations, but this is the easiest to reproduce I guess. I think this is related to: https://github.com/linuxdeepin/dde-dock/issues/101

**Comment by jouyouyun** [comment]
@electricface @haruyukilxz

**Comment by igormcoelho** [comment]
Please, its so annoying xD

**Comment by core01** [comment]
Hi all, I guess this problem is also related to https://github.com/linuxdeepin/dde-dock/issues/49

**Comment by igormcoelho** [comment]
I confirm this was solved on Deepin 15.8 :fireworks: 
I don't know how they did it, but now when dde-dock freezes (it still freezes sometimes) the system resets it after a few seconds... it's like a watchdog that's running somewhere. Thanks for that!

**Comment by KewlToyZ** [comment]
daemon/dock[2007]: dock_manager_xevent.go:154: x.Error: 3 (BadWindow), sequence: 302, resource id: 29360566, major code: 2 (ChangeWindowAttributes), minor code: 0
I'm still seeing it in UbuntuDDE 20.04 in 2020?
